script.js

```diff
@@ -0,0 +1,578 @@
+        ArrowRight: '&#8594;',
+        ArrowDown: '&#8595;',
+        ArrowLeft: '&#8592;',
+      },
+    },
+  },
+
+
+  // method for creating new elements
+  createNewElement(tag, tagClass, tagText) {
+    const elm = document.createElement(tag);
+    elm.className = tagClass;
+    elm.innerHTML = tagText;
+    return elm;
+  },
+
+
+  // method for create KexBox element, arguments:array keys, array function keys
+  createKeyBox(arrKeys, arrFuncKeys) {
+    const keyBox = this.createNewElement('div', 'keyboard', '');
+
+    for (let i = 0; i < arrKeys.length; i += 1) {
+      const keysRow = this.createNewElement('div', 'keys-row', '');
+
+      for (let j = 0; j < arrKeys[i].length; j += 1) {
+        const key = arrKeys[i][j];
+        let classesStr = `key ${key}`.toLowerCase();
+
+        if (arrFuncKeys.includes(key)) classesStr = `${classesStr} key--function`;
+
+        const keyElm = this.createNewElement('button', classesStr, '');
+        keyElm.setAttribute('data-key-code', key);
+        keysRow.append(keyElm);
+      }
+      keyBox.append(keysRow);
+    }
+    return keyBox;
+  },
+
+
+  // method for make keyboard language layout, argument:('en' or 'ru')
+  setLanguageLayout(languageToggle) {
+    const arrKeys = this.elements.keyBox.querySelectorAll('.key');
+    for (let i = 0; i < arrKeys.length; i += 1) {
+      const elem = arrKeys[i];
+      const keySymbol = this.language[languageToggle].lowerCase[elem.dataset.keyCode];
+      elem.innerHTML = keySymbol;
+    }
+    this.inputAria.languageToggle = languageToggle;
+    localStorage.setItem('virtualKeyboardLang', languageToggle);
+  },
+
+
+  // method set uppercase keyboard when downKey Shift, argument:(not)
+  setShiftDownLayout() {
+    const arrKeys = this.elements.keyBox.querySelectorAll('.key:not(.key--function)');
+    for (let i = 0; i < arrKeys.length; i += 1) {
+      const elem = arrKeys[i];
+      const keySymbol = this.language[this.inputAria.languageToggle].shift[elem.dataset.keyCode];
+      elem.innerHTML = keySymbol;
+    }
+  },
+
+
+  // method set lowercase keyboard when upKey Shift, argument:(not)
+  setShiftUpLayout() {
+    const arrKeys = this.elements.keyBox.querySelectorAll('.key');
+    for (let i = 0; i < arrKeys.length; i += 1) {
+      const elem = arrKeys[i];
+      const keySymbol = this.language[this.inputAria.languageToggle]
+        .lowerCase[elem.dataset.keyCode];
+      elem.innerHTML = keySymbol;
+    }
+  },
+
+
+  // method switch lowercase keyboard when upKey CapsLock, argument:(not)
+  setCapsLock() {
+    const arrKeys = this.elements.keyBox.querySelectorAll('.key:not(.key--function)');
+
+    for (let i = 0; i < arrKeys.length; i += 1) {
+      const elem = arrKeys[i];
+      const keySymbol = elem.innerHTML.toUpperCase();
+      elem.innerHTML = keySymbol;
+    }
+  },
+
+
+  // method switch Language Layout for keyboard, argument:(not)
+  switchLanguageLayout() {
+    if (this.inputAria.languageToggle === 'en') {
```

Comment:
I think it is better to store such strings into a constant variable

Reply:
where the best place is constant variables in my case? I have one big object KeyBoard.

Reply:
If you have many constants you can move them to a separate file `constants.js` or introduce them at the top of current file(function/block)

Reply:
Done